### PR TITLE
Stability circles

### PR DESCRIPTION
**Pull request by keikawa**
This pull request adds stability circle calculation methods related to #362.

### CODE REVIEW

File: skrf/network.py [hunk 1]
```diff
@@ -4267,6 +4272,177 @@ def vswr_active(self, a: npy.ndarray) -> npy.ndarray:
...
+            raise ValueError("Stability circle is only defined for two ports")
+        
+        if npoints <= 0:
+            raise ValueError("npoints must be a positive integer")
+
+        # Calculate the determinant of the scattering matrix
+        D = self.s[:, 0, 0] * self.s[:, 1, 1] - self.s[:, 0, 1] * self.s[:, 1, 0]
+
+        # Calculate the center and radius of the source stability circle
+        ssc_center = (self.s[:, 0, 0] - self.s[:, 1, 1].conjugate() * D).conjugate() / (npy.abs(self.s[:, 0, 0]) ** 2 - npy.abs(D) ** 2)
```
Comment (FranzForstmayr) [inline skrf/network.py:4432]: This two lines are the only differences between `load` and `source`  circles, isn't it?
Would it be better to make a generic `stability_circle` method where one can choose `source` or `load` via an attribute?

Comment (keikawa) [reply skrf/network.py]: Thank you for your suggestion. I have combined these two methods into one.

**Review by jhillairet — APPROVED** [review]
Nice addition, thank you. 

Do you think you could write an associated notebook for the tutorial part of the documentation, to explain what are stability circles and how to use it?

**Comment by keikawa** [comment]
That sounds good. 
I'm planning to start by implementing the gain circle calculation methods and then move on to creating a notebook, because stability circles are typically used with gain circles, as requested in #362.
But, unfortunately, I'm currently busy this month due to preparations for conferences, so I'd prefer to send other pull requests for these tasks.

**Pull request merged**